### PR TITLE
Add parameter_set method, and remove .param_set calls

### PR DESCRIPTION
**Pull request by charleskawczynski**
### Description

This PR adds `parameter_set` to `BalanceLaws` and uses this in favor of `.param_set`. If we create/use enough getter wrappers, then we won't need to duplicate code in #2100.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: experiments/AtmosLES/dycoms.jl [hunk 1]
```diff
@@ -108,9 +108,10 @@ function flux(::DYCOMSRadiation{Energy}, atmos, args)
     # Constants
     upward_flux_from_cloud = m.F_0 * exp(-aux.∫dnz.radiation.attenuation_coeff)
     upward_flux_from_sfc = m.F_1 * exp(-aux.∫dz.radiation.attenuation_coeff)
+    param_set
```
Comment (ilopezgp) [inline experiments/AtmosLES/dycoms.jl:111]: param_set = parameter_set(atmos) ?

File: src/Atmos/Model/bc_energy.jl [hunk 2]
```diff
@@ -35,8 +35,9 @@ function atmos_energy_boundary_state!(
 )
     @unpack aux⁻, state⁻, t = args
     FT = eltype(aux⁻)
-    _T_0::FT = T_0(atmos.param_set)
-    _cv_d::FT = cv_d(atmos.param_set)
+    param_set
```
Comment (ilopezgp) [inline src/Atmos/Model/bc_energy.jl:38]: I see this line several times. Is it supposed to define the param_set from the getter, but the RHS is missing?

Comment (charleskawczynski) [reply src/Atmos/Model/bc_energy.jl]: Thanks, fixed.

File: experiments/AtmosLES/dycoms.jl (hunk 1 above)
Comment (charleskawczynski) [reply experiments/AtmosLES/dycoms.jl]: Thanks, fixed.

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Review by kpamnany — COMMENTED** [review]
I'm not seeing what this accomplishes?

**Comment by charleskawczynski** [comment]
> I'm not seeing what this accomplishes?

It's still a WIP, but the basic idea is that we'd like to be able to incrementally assemble the atmos model (see #2100). We'll need to change the internal structure to do that without duplicating fields. This PR will allow us to restructure the `AtmosModel` to have `atmos.physics.param_set` (still haven't made that change yet). We'll need to do the same with all other atmos properties, I figure this is the most wide-spread one, though.

**Review by kpamnany — APPROVED** [review]
With the caveat that many getter functions will make the interface more wall-like, this does address the problem you're trying to solve.

**Pull request merged**